brainiak/searchlight/searchlight.py

```diff
@@ -0,0 +1,353 @@
+        """ Distribute tasks dynamically to ranks 1-size (worker)
+
+        Parameters
+        ----------
+        tasks: searchlight centers, a list of tuples
+
+        data: A list of 4D numpy.ndarray objects which must have at least one
+        element. The computation is applied to this data.
+
+        mask: A 3D numpy.ndarray object with type np.bool indicating which
+        voxels are active in the searchlight operation.
+
+        bcast_var: An object of any pickle-able type which is included
+        as a parameter to each invocation of the user-specified function
+
+        Returns
+        -------
+        None
+
+        """
+
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+
+        # Request task (empty result)
+        comm.send((rank, None), dest=0)
+
+        # Infinite loop
+        while True:
+
+            # Receive task
+            (idx, d, m) = comm.recv(source=0)
+
+            # If task is empty then break loop
+            if(idx == -1):
+                break
+
+            result = self.fn(d, m, bcast_var)
+
+            # Send result
+            comm.send((rank, (idx, result)), dest=0)
+
+    def _dynamic_tasking(self, tasks, data, mask, bcast_var):
+        """ Distribute tasks dynamically to ranks 1-size
+
+        Parameters
+        ----------
+        tasks: searchlight centers, a list of tuples
+
+        data: A list of 4D numpy.ndarray objects which must have at least one
+        element. The computation is applied to this data.
+
+        mask: A 3D numpy.ndarray object with type np.bool indicating which
+        voxels are active in the searchlight operation.
+
+        bcast_var: An object of any pickle-able type which is included
+        as a parameter to each invocation of the user-specified function
+
+        Returns
+        -------
+        results: A 3D numpy.ndarray object with the outputs of each
+        function invocation
+
+        """
+
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+
+        bcast_var = comm.bcast(bcast_var, root=0)
+
+        results = []
+
+        if rank == 0:
+            results = self._do_master(tasks, data, mask, bcast_var)
+
+        if rank != 0:
+            self._do_worker(tasks, data, mask, bcast_var)
+
+        return results
+
+    def fit_transform(self, X, y=None):
+        """ Run searchlight according to mask
+
+        Applies a function to each voxel present in the mask.
+
+        Parameters
+        ---------
+        X: A tuple containing two objects:
+
+          data: A list of 4D numpy.ndarray objects which must have at least
+          one element. The computation is applied to this data. This needs
+          only to be present on rank 0.
+
+          mask: A 3D numpy.ndarray object with type np.bool indicating which
+          voxels are active in the searchlight operation. This needs
+          only to be present on rank 0
+
+        y: An object of any pickle-able type which is included
```

Comment:
Any parameter that is **unrelated** to the data should be passed in the constructor.
Please, consider moving it to the constructor. Otherwise, you need an additional parameter.
"y" is defined as the labels of the data. If there are no labels, then should always be None.